exercises/src/main/scala/fpinscala/laziness/Stream.scala

```diff
@@ -17,20 +27,95 @@ trait Stream[+A] {
     case Empty => None
     case Cons(h, t) => if (f(h())) Some(h()) else t().find(f)
   }
-  def take(n: Int): Stream[A] = ???
 
-  def drop(n: Int): Stream[A] = ???
+  //5.2
+  def take(n: Int): Stream[A] = this match {
+    case _ if n == 0 => empty
+    case Cons(h, _) if n == 1 => cons(h(), empty)
```

Comment:
What would happen if you didn't have this line?

Reply:
On second look, I actually realized that this line is actually required to cover this scenario without letting it go through the next case, the behavior wouldn't be exactly the same. I'm guessing you added it because you saw the issue with lazy vs strict for the tail?

Reply:
I thought without that line the function always evaluates the tail one extra time.  I'm not sure if this is the issue you're referring to

Reply:
Yes, that's what I meant.